src/Type/InspectorTypeExtension.php

```diff
@@ -234,9 +233,9 @@ private function specifyAssertAllHaveKey(MethodReflection $staticMethodReflectio
             $keyTypes[] = new HasOffsetType(new ConstantStringType($key));
         }
 
-        $newArrayType = new ArrayType(
+        $newArrayType = new IterableType(
             new MixedType(true),
-            new ArrayType(TypeCombinator::intersect(new MixedType(), ...$keyTypes), new MixedType(true)),
+            new ArrayType(TypeCombinator::union(new MixedType(), ...$keyTypes), new MixedType(true)),
```

Comment:
Intersect gives us `hasOffset` but it's lost in `union`... maybe because we keep `MixedType` or `MixedType` is not `true`? If we have `HasOffsetType` of string... can we even have `MixedType` anymore?

So 

```
$keyTypes[] = new IntegerType();
$keyTypes[] = new StringType();

new ArrayType(TypeCombinator:: intersect(...$keyTypes), new MixedType(true)),
```

Reply:
Removing `MixedType` still causes `'*NEVER*`